libvirt/resource_libvirt_domain.go

```diff
@@ -267,6 +289,25 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if kernel, ok := d.GetOk("kernel"); ok {
+		domainDef.OS.Kernel = kernel.(string)
+	}
+	if initrd, ok := d.GetOk("initrd"); ok {
+		domainDef.OS.Initrd = initrd.(string)
+	}
+
+	cmdlinesCount := d.Get("cmdline.#").(int)
+	var cmdlineArgs []string
+	for i := 0; i < cmdlinesCount; i++ {
+		cmdlineKey := fmt.Sprintf("cmdline.%d", i)
+		cmdlineMap := d.Get(cmdlineKey).(map[string]interface{})
+		for k, v := range cmdlineMap {
```

Comment:
You could also shorten this if you wanted to. The variables are only used once anyway.
```go
for k, v := range d.Get(fmt.Sprintf("cmdline.%d", i)).(map[string]interface{}) {
```
IMHO the code is still pretty clear.

Reply:
good